base/show.jl

```diff
@@ -1094,30 +1111,41 @@ function show_datatype(io::IO, x::DataType, wheres::Vector{TypeVar}=TypeVar[])
         return
     elseif isnamedtuple
         syms, types = parameters
-        first = true
         if syms isa Tuple && types isa DataType
             print(io, "@NamedTuple{")
-            for i in 1:length(syms)
-                if !first
-                    print(io, ", ")
-                end
-                print(io, syms[i])
-                typ = types.parameters[i]
-                if typ !== Any
-                    print(io, "::")
-                    show(io, typ)
-                end
-                first = false
-            end
+            show_at_namedtuple(io, syms, types)
             print(io, "}")
             return
         end
+    elseif get(io, :simplify_kwstype, false) && kwsnt !== nothing
```

Comment:
I enabled the printing with the `@Kwargs`-macro syntax only within stack trace view. Please let me know if you think it is more reasonable to enable the printing in the other situations.

The main reasoning behind the current choice is that `Base.Pairs` is a general data structure and its usage is not restricted to represent keyword arguments.